### ISSUE TITLE
ERR_UNKNOWN_ENCODING throw when call.write(request, metadata)

### ISSUE THREAD

**Issue by feuyeux** [grpc/grpc-node#1849]
### Problem description
When to migrate from grpc to grpc-js, I met the below issue:
if send metadata through `call.write` in client side, it will throw the below issue:
```bash
2021-07-15T12:50:04.245Z [info] TalkBidirectional:4,NODEJS
node:internal/streams/writable:293
      throw new ERR_UNKNOWN_ENCODING(encoding);
      ^

TypeError [ERR_UNKNOWN_ENCODING]: Unknown encoding: Metadata { internalRepr: [Map], options: {} }
    at new NodeError (node:internal/errors:363:5)
    at _write (node:internal/streams/writable:293:13)
    at ClientDuplexStreamImpl.Writable.write (node:internal/streams/writable:334:10)
```

client code: 
```javascript
requests.forEach(request => {
        logger.info("TalkMoreAnswerOne:" + request)
        // call.write(request, metadata)
        call.write(request)
    })
```

https://github.com/AliyunContainerService/hello-servicemesh-grpc/blob/9f5c05222a12e5f82d24a30dfdb1d49a71cc4b5c/grpc/hello-grpc-nodejs/proto_client.js#L117

server side code:
https://github.com/AliyunContainerService/hello-servicemesh-grpc/blob/9f5c05222a12e5f82d24a30dfdb1d49a71cc4b5c/grpc/hello-grpc-nodejs/proto_server.js#L215

### Reproduction steps
https://github.com/AliyunContainerService/hello-servicemesh-grpc/blob/main/grpc/hello-grpc-nodejs/README.md

### Environment
 - OS name, version and architecture: macos big sur 11.4
 - Node version: v16.4.2

**Comment by murgatroid99** [comment]
That is not where metadata goes. You need to pass metadata as an argument before the callback when initially calling the method, like you are doing [here](https://github.com/AliyunContainerService/hello-servicemesh-grpc/blob/9f5c05222a12e5f82d24a30dfdb1d49a71cc4b5c/grpc/hello-grpc-nodejs/proto_client.js#L100)

**Comment by feuyeux** [comment]
> That is not where metadata goes. You need to pass metadata as an argument before the callback when initially calling the method, like you are doing [here](https://github.com/AliyunContainerService/hello-servicemesh-grpc/blob/9f5c05222a12e5f82d24a30dfdb1d49a71cc4b5c/grpc/hello-grpc-nodejs/proto_client.js#L100)

Thanks for your reply!
Firstly, this behavior by using gprc-js is same with using grpc, and it works with grpc lib.
Secondly, send metadata as the seconde param with callback is Unary RPC pattern, and my issue happens in Client streaming RPC and Bidirectional streaming RPC, which send request more than once.
Fianlly, the stub api defined as a template in codegen processing, I cannot change that.

![image](https://user-images.githubusercontent.com/769711/125887160-ae8c639a-a657-4a79-966b-ec09147ca974.png)
![image](https://user-images.githubusercontent.com/769711/125887209-ee5193a6-13ce-4cb3-809e-9b021d8b5d81.png)

**Comment by murgatroid99** [comment]
First, the `grpc` package may allow you to pass a metadata object as the second argument to `write` without throwing an error, but I am absolutely sure that it will never actually send metadata passed that way.

Second, metadata can be passed as an argument before the callback (or at the end if there is no callback) for any kind of method. That is the only way to send metadata from the client.

More specifically, the argument order is

 - Message, if unary or server streaming, otherwise nothing
 - Optional metadata
 - Optional call options
 - Callback, if unary or client streaming, otherwise nothing

Third, the `landing_grpc_pb.js` file does not define any method signatures at all, so I am unsure how you think it says that any particular method signature is correct or incorrect.

**Comment by feuyeux** [comment]
@murgatroid99 Hi Michael, thanks a lot!
In the end, I found it was my mistake, and it's not an issue.

1. when I double check landing_grpc_pb.js, it's clear and no method signatures which in my mind, the above pictures show that IDE "cheated" me :)
2. I was not familiar the way to set the params of the method is dynamic, in nodejs, relative to java or golang.

When I fixed my code follow your suggetstion, it works! And I think grpc-js is more standard than grpc, it provides enough flexibility, and while, not  allow the bad behaviour codes.

However,  it would be better if there's a full scope demo or offical doc. Since some people like me that nodejs is not our first language, but we need to support this popular language with our scenario, e.g. servicemesh with gRPC.
